privilege/privileges/privileges_test.go

```diff
@@ -1466,10 +1464,8 @@ func (s *testPrivilegeSuite) TestSecurityEnhancedModeInfoschema(c *C) {
 
 	// That is unless we have the RESTRICTED_TABLES_ADMIN privilege
 	tk.Se.Auth(&auth.UserIdentity{
-		Username:     "uroot2",
-		Hostname:     "localhost",
-		AuthUsername: "uroot",
-		AuthHostname: "%",
+		Username: "uroot2",
+		Hostname: "localhost",
```

Comment:
The username doesn't match the AuthUsername, which looks like a problem waiting to happen. I wrote this test but must have missed it :(